pytest_instafail.py

```diff
@@ -21,13 +21,42 @@ def pytest_addoption(parser):
             "default)."
         )
     )
+    try:
+        parser.addini(
+            "instafail", default=False, type="bool",
+            help=(
+                "show failures and errors instantly as they occur (disabled by "
+                "default)."
+            )
+        )
+    except AssertionError:
+        # This is pytest < 2.9.0
+        parser.addini(
+            "instafail", default=False, type=None,
+            help=(
+                "show failures and errors instantly as they occur (disabled by "
+                "default)."
+            )
+        )
 
 
 @pytest.mark.trylast
 def pytest_configure(config):
     if hasattr(config, 'slaveinput'):
         return  # xdist slave, we are already active on the master
-    if config.option.instafail:
+
+    ini_instafail_setting = config.getini('instafail')
+    if not isinstance(ini_instafail_setting, bool):
+        # Pytest < 2.9.0
+        ini_instafail_setting = ini_instafail_setting.lower().strip()
+        if ini_instafail_setting in ('y', 'yes', 't', 'true', 'on', '1'):
+            ini_instafail_setting = True
+        elif val in ('n', 'no', 'f', 'false', 'off', '0'):
+            ini_instafail_setting = False
+        else:
+            raise ValueError("invalid truth value %r" % (ini_instafail_setting,))
```

Comment:
I leave that option to the owner/maintainer. I can however update it if desired.